encoding/textdecoder-labels.html

```diff
@@ -4,33 +4,32 @@
 <script src="/resources/testharnessreport.js"></script>
 <script src="resources/encodings.js"></script>
 <script>
-var tests = [];
-setup(function() {
-  var whitespace = [' ', '\t', '\n', '\f', '\r'];
-  encodings_table.forEach(function(section) {
-    section.encodings.filter(function(encoding) {
-      return encoding.name !== 'replacement';
-    }).forEach(function(encoding) {
-      var name = encoding.name;
-      encoding.labels.forEach(function(label) {
-        tests.push([label, encoding.name]);
+encodings_table.forEach(function(section) {
+  section.encodings.filter(function(encoding) {
+    return encoding.name !== 'replacement';
+  }).forEach(function(encoding) {
+    encoding.labels.forEach(function(label) {
+      test(function(t) {
+        assert_equals(
+          new TextDecoder(label).encoding, encoding.name,
+          'label for encoding should match');
+        assert_equals(
+          new TextDecoder(label.toUpperCase()).encoding, encoding.name,
+          'label matching should be case-insensitive');
+        var whitespace = [' ', '\t', '\n', '\f', '\r'];
```

Comment:
Seems better to define this outside of the loops?

Anyway, looks okay to me overall. This is your test and @Ms2ger modified it a bit. I don't really care how it's formatted personally, as long as it still tests the same things and we get the same relative failure rates.